### DISCUSSION TITLE
Multiline text input / mini editor

### DISCUSSION THREAD

**Discussion by leonjza** [Textualize/textual#1035] [Q&A] [answered]
Hello! First, thanks for this amazing library! I am having a lot of fun with it!

I was wondering if there was an example somewhere that has a multiline code editor Widget (optionally with syntax highlighting) anywhere?

**Comment by leonjza** [comment]
To add some more context, I've played with the `Input` widget, but things in my way at the moment is the inability to unbind `enter` so that I can have multiple lines, and a way to process the contents such that I could apply syntax highlighting to it afterwards. Alternatives I'm considering is building a new widget, but it sounds useful enough that maybe something already exists.

**Comment by davep** [comment, marked as answer]
Textual doesn't have a builtin multi-line editor at the moment, but a `TextArea` editor is [on the roadmap](https://textual.textualize.io/roadmap/#widgets).

**Reply by giohappy** [reply]
Do you already have a tentative / indicative date when this will be available, even just for testing a development preview?

**Reply by giohappy** [reply]
I just found the `simple-multiline-text` branch. I guess it's where it is being developed.

**Reply by cargilcm** [reply]
I don't see anywhere on [that branch](https://github.com/Textualize/textual/tree/simple-multiline-text) where this widget is being developed. @giohappy : happy to learn if I missed it perhaps? Thanks

**Reply by giohappy** [reply]
@cargilcm https://github.com/Textualize/textual/tree/simple-multiline-text

**Reply by cargilcm** [reply]
I see the example script is just called [simple.py. I DID MISS IT](https://github.com/Textualize/textual/blob/simple-multiline-text/examples%2Fsimple.py)!

**Reply by willmcgugan** [reply]
We've had a TextArea widget for some time. https://textual.textualize.io/widget_gallery/#textarea

**Comment by leonjza** [comment]
Ah sorry I missed that, but thank you for the information! Excited for it!

**Reply by davep** [reply]
No apology needed, it's an obvious thing to go looking for. :-)

**Reply by cargilcm** [reply]
I see that there's a widget Text area that was developed but it doesn't have built-in horizontal scrolling like RichLog. Wouldn't this be a nice feature? I'd be willing to help if I could ask a few pointers here and there, and thank you guys for the coolness found in textual ;)

**Reply by davep** [reply]
@cargilcm Sounds like you might have missed the [`soft_wrap`](https://textual.textualize.io/widgets/text_area/#textual.widgets._text_area.TextArea(soft_wrap)) parameter, perhaps?

**Reply by cargilcm** [reply]
This helps so much!
